src/mechanics_ecmech.cpp

```diff
@@ -833,7 +338,286 @@ void ExaCMechModel::ModelSetup(const int nqpts, const int nelems, const int spac
+
+   ind_dp_eff = ecmech::evptn::iHistA_shrateEff;
+   ind_eql_pl_strain = ecmech::evptn::iHistA_shrEff;
+   ind_num_evals = ecmech::evptn::iHistA_nFEval;
+   ind_dev_elas_strain = ecmech::evptn::iHistLbE;
+   ind_quats = ecmech::evptn::iHistLbQ;
+   ind_hardness = ecmech::evptn::iHistLbH;
+
+   ind_gdot = mat_model->iHistLbGdot;
+   // This will always be 1 for this class
+   num_hardness = mat_model->nH;
+   // This will always be 12 for this class
+   num_slip = mat_model->nslip;
+   // The number of vols -> we actually only need to save the previous time step value
+   // instead of all 4 values used in the evalModel. The rest can be calculated from
+   // this value.
+   num_vols = 1;
+   ind_vols = ind_gdot + num_slip;
+   // The number of internal energy variables -> currently 1
+   num_int_eng = ecmech::ne;
+   ind_int_eng = ind_vols + num_vols;
+
+   // Params start off with:
+   // initial density, heat capacity at constant volume, and a tolerance param
+   // Params then include Elastic constants:
+   // c11, c12, c44 for Cubic crystals
+   // Params then include the following:
+   // reference shear modulus, reference temperature, g_0 * b^3 / \kappa where b is the
+   // magnitude of the burger's vector and \kappa is Boltzmann's constant, Peierls barrier,
+   // MTS curve shape parameter (p), MTS curve shape parameter (q), reference thermally activated
+   // slip rate, reference drag limited slip rate, drag reference stress, slip resistance const (g_0),
+   // slip resistance const (s), dislocation density production constant (k_1),
+   // dislocation density production constant (k_{2_0}), dislocation density exponential constant,
+   // reference net slip rate constant, reference relative dislocation density
+   // Params then include the following:
+   // the Grüneisen parameter, reference internal energy
+
+   // Opts and strs are just empty vectors of int and strings
+   std::vector<double> params;
+   std::vector<int> opts;
+   std::vector<std::string> strs;
+   // 16 terms come from hardening law, 3 terms come from elastic modulus, 4 terms are related to EOS params,
+   // 1 terms is related to solving tolerances == 24 total params
+   MFEM_ASSERT(matProps->Size() == 24, "Properties need 24 parameters for FCC MTS like hardening model");
+
+   for (int i = 0; i < matProps->Size(); i++) {
+      params.push_back(matProps->Elem(i));
+   }
+
+   // We really shouldn't see this change over time at least for our applications.
+   mat_model_base->initFromParams(opts, params, strs, accel);
+   //
+   mat_model_base->complete();
+
+   std::vector<double> histInit;
+   {
+      std::vector<std::string> names;
+      std::vector<bool>        plot;
+      std::vector<bool>        state;
+      mat_model_base->getHistInfo(names, histInit, plot, state);
+   }
+   init_state_vars(_q_matVars0, histInit);
+}
+
+void KinKMBalDDFCCModel::init_state_vars(mfem::QuadratureFunction *_q_matVars0, std::vector<double> hist_init)
```

Comment:
I have _not_ managed to grok much of this part of the code, and am really just skimming through. But I am a bit surprised to see details such as the KinKMBalDDFCC name finding their way this far up the call stack. Are there things that this level that are actually specific to the details of the material model being employed?

Reply:
So, I have a regular ExaCMech base class that these classes are derived off and for all intents and purposes that's where the meat of things occur. However, I decided it would be easier to have child classes that contain the actual templated material classes within them where I can then correctly instantiate the material models. For example the ```KinKMBalDDFCC``` class really makes use of the already defined ```matModelEvptn_FCC_B``` within ```ExaCMech```. However, I didn't like that name and thought the ```KinKMBalDDFCC``` one would be more descriptive of the actual model being used for this class.

If you'd like we could talk more offline if the above wasn't super clear about what these classes are actually for.